### PR TITLE
Add relay

### PR DESCRIPTION
**Pull request by lucasbesen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,19 +4,25 @@
...
-    "test": "jest"
+    "test": "jest",
+    "relay": "relay-compiler --src ./src --schema ./data/schema.graphql"
   },
   "dependencies": {
     "react": "16.2.0",
     "react-native": "0.53.3",
     "react-navigation": "^1.2.1",
+    "react-relay": "^1.5.0",
+    "relay": "^0.8.0-1",
```
Comment (jgcmarins) [inline package.json]: you do not need this dep

File: src/screens/HomeScreen.js [hunk 2]
```diff
@@ -1,28 +1,51 @@
...
-      </HomeView>
+      <QueryRenderer
+        environment={ env }
+        query={ query }
+        render={({error, props}) => {
+          if (error) {
+            return (<Error message={error.message} />);
+          } else if (props) {
+            return (
+              <HomeView>
```
Comment (jgcmarins) [inline src/screens/HomeScreen.js]: this can be another component

**Review by jgcmarins — APPROVED** [review]
Looks good.
You can try implement a generic component that handles QueryRender and returns a component received as `props`.

File: src/components/Error.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+
+const Error = (props) => (
```
Comment (lXSPandora) [inline src/components/Error.js:4]: very nice i like stateless components

File: src/queries/AllStudentsQuery.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  query AllStudentsQuery {
+    allStudents {
+      _id
+      name
+      description
+    }
+  }
+`;
+
+const StudentsQuery = (componentProps) => (
```
Comment (lXSPandora) [inline src/queries/AllStudentsQuery.js:19]: nice!

**Review by lXSPandora — APPROVED** [review]
Great Job 💯

**Pull request merged**